src/main/kotlin/racingcar/application/parser/StringToIntParser.kt

```diff
@@ -0,0 +1,18 @@
+package racingcar.application.parser
+
+import racingcar.domain.error.ErrorCode.ATTEMPT_COUNT_MUST_BE_GREATER_THAN_ZERO
+import racingcar.domain.error.ErrorCode.INVALID_ATTEMPT_COUNT_NUMBER
+
+class StringToIntParser : Parser<Int> {
+    override fun parse(input: String): Int {
+        try {
+            val convertInt = input.toInt()
+            if (convertInt <= 0) {
+                throw IllegalArgumentException(ATTEMPT_COUNT_MUST_BE_GREATER_THAN_ZERO)
+            }
+        } catch (e: NumberFormatException) {
+            throw IllegalArgumentException(INVALID_ATTEMPT_COUNT_NUMBER)
+        }
+        return input.toInt()
```

Comment:
현재 `convertInt<=0`일때 throw를 하고 try-catch로 잡아서 다시 던지는 부분이 조금 비효율적이라는 생각이 듭니다
`toInt`대신 `toIntOrNull`을 사용하면 try-catch를 없이도 구현할 수 있을 것 같습니다!
```suggestion
    override fun parse(input: String): Int {
        val convertInt = input.toIntOrNull() ?: throw IllegalArgumentException(INVALID_ATTEMPT_COUNT_NUMBER)
        if (convertInt <= 0) throw IllegalArgumentException(ATTEMPT_COUNT_MUST_BE_GREATER_THAN_ZERO)
        
        return convertInt 
```

또한 require를 활용하면 아래와 같이 작성할 수도 있을 것 같습니닷

```
val convertInt = input.toIntOrNull()
require(convertInt != null) { INVALID_ATTEMPT_COUNT_NUMBER }
require(convertInt > 0) { ATTEMPT_COUNT_MUST_BE_GREATER_THAN_ZERO }
```

Reply:
> 현재 `convertInt<=0`일때 throw를 하고 try-catch로 잡아서 다시 던지는 부분이 조금 비효율적이라는 생각이 듭니다 `toInt`대신 `toIntOrNull`을 사용하면 try-catch를 없이도 구현할 수 있을 것 같습니다!
> 
> 또한 require를 활용하면 아래와 같이 작성할 수도 있을 것 같습니닷
> 
> ```
> val convertInt = input.toIntOrNull()
> require(convertInt != null) { INVALID_ATTEMPT_COUNT_NUMBER }
> require(convertInt > 0) { ATTEMPT_COUNT_MUST_BE_GREATER_THAN_ZERO }
> ```
코드로도 알려주시고 피드백 감사합니다!!